packages/components/src/text-input/hooks/use-text-input.js

```diff
@@ -0,0 +1,194 @@
+/**
+ * External dependencies
+ */
+import { cx } from 'emotion';
+import TextareaAutosize from 'react-autosize-textarea';
```

Comment:
I've added this to `package.json` 

Is there an alternative or are we okay with adding more dependencies here?

Reply:
It should be okay to add this dependency: https://bundlephobia.com/package/react-autosize-textarea@7.1.0 it's pretty small.